orderly_web/config.py

```diff
@@ -69,6 +78,11 @@ def get_container(self, name):
         with docker_client() as cl:
             return cl.containers.get(self.containers[name])
 
+    def resolve_secrets(self):
+        vault_client = self.vault.client()
```

Comment:
Seems like secret resolving should happen when the config class is initialised, rather than in a separate call, as you'd never not want to resolve secrets.

Relatedly, should using a vault for secrets be optional?

Reply:
I do similar things with orderly secrets and it's nice to be able to load things without resolving secrets.  Aside from anything else, testing this would be much harder if we resolved secrets straight away because we need to know the vault address and we'd have to write out configurations (though the work for patching configurations will make that nicer, but I wanted to keep that out of this PR)

I don't understand your second question - vault is optional, but were you imagining an alternative way of resolving secrets?

Reply:
No I just meant is the vault optional full stop bc it looked to me like it was required by the code (see my other comment, I misunderstood where the exception was being thrown)